scotch build issue on bg-q (cross-compiling env)

On `bg-q` building `scotch_6.0.4` gives : 

```bash

$ spack install -v --keep-stage scotch -shared
.........
/gpfs/mira-home/kumbhar/spack/lib/spack/env/xl/xlc -O3 -DCOMMON_RANDOM_FIXED_SEED -DSCOTCH_DETERMINISTIC -DSCOTCH_RENAME -qstrict -qlanglvl=extc99 -s -D_ALL_SOURCE -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_PTHREAD  -c vgraph_separate_st.c -o vgraph_separate_st.o
./dummysizes library.h scotch.h
./dummysizes library_f.h scotchf.h
make[2]: *** [scotchf.h] Illegal instruction (core dumped)
make[2]: *** Waiting for unfinished jobs....
```
Obvious issue : trying to run compute node exe on login node. Note that the provided `Makefile.inc` build the package properly : 
```
Make.inc/Makefile.inc.ppca2_ibm_bgq
```

P.S. I am trying to build set of packages on bg-q in cross-compiling environment. Will create issues as I encounter the failures. Will update these later possibly with patches.

--- a/var/spack/repos/builtin/packages/scotch/package.py
+++ b/var/spack/repos/builtin/packages/scotch/package.py
@@ -144,6 +144,9 @@ class Scotch(Package):
             ldflags.append('-L%s -lz' % (self.spec['zlib'].prefix.lib))
 
         cflags.append('-DCOMMON_PTHREAD')
+
+        # NOTE: bg-q platform needs -lpthread (and not -pthread)
+        # otherwise we get illegal instruction error during runtime
         if self.spec.satisfies('platform=darwin'):
             cflags.append('-DCOMMON_PTHREAD_BARRIER')
             ldflags.append('-lm -pthread')

--- a/var/spack/repos/builtin/packages/scotch/package.py
+++ b/var/spack/repos/builtin/packages/scotch/package.py
@@ -56,6 +56,10 @@ class Scotch(Package):
     depends_on('mpi', when='+mpi')
     depends_on('zlib', when='+compression')
 
+    # NOTE: In cross-compiling environment parallel build
+    # produces weired linker errors.
+    parallel = False
+
     # NOTE: Versions of Scotch up to version 6.0.0 don't include support for
     # building with 'esmumps' in their default packages.  In order to enable
     # support for this feature, we must grab the 'esmumps' enabled archives
@@ -143,6 +147,8 @@ class Scotch(Package):
         if self.spec.satisfies('platform=darwin'):
             cflags.append('-DCOMMON_PTHREAD_BARRIER')
             ldflags.append('-lm -pthread')
+        elif self.spec.satisfies('platform=bgq'):
+            ldflags.append('-lm -lrt -lpthread')
         else:
             ldflags.append('-lm -lrt -pthread')
 

--- a/var/spack/repos/builtin/packages/scotch/package.py
+++ b/var/spack/repos/builtin/packages/scotch/package.py
@@ -57,7 +57,7 @@ class Scotch(Package):
     depends_on('zlib', when='+compression')
 
     # NOTE: In cross-compiling environment parallel build
-    # produces weired linker errors.
+    # produces weird linker errors.
     parallel = False
 
     # NOTE: Versions of Scotch up to version 6.0.0 don't include support for

